src/Object/Photo.php

```diff
@@ -653,417 +758,244 @@ public function store($uid, $cid, $rid, $filename, $album, $scale, $profile = 0,
+			$filesize = strlen($img_str);
+	
+			if (function_exists("getimagesizefromstring")) {
+				$data = getimagesizefromstring($img_str);
+			} else {
+				$tempfile = tempnam(get_temppath(), "cache");
+	
+				$a = get_app();
+				$stamp1 = microtime(true);
+				file_put_contents($tempfile, $img_str);
+				$a->save_timestamp($stamp1, "file");
+	
+				$data = getimagesize($tempfile);
+				unlink($tempfile);
+			}
+	
+			if ($data) {
+				$data["size"] = $filesize;
+			}
+	
+			Cache::set($url, $data);
+		}
+	
+		return $data;
 	}
 
-	if ($width > $max && $height > $max) {
-
-		// very tall image (greater than 16:9)
-		// constrain the width - let the height float.
-
-		if ((($height * 9) / 16) > $width) {
-			$dest_width = $max;
-			$dest_height = intval(($height * $max) / $width);
-		} elseif ($width > $height) {
-			// else constrain both dimensions
-			$dest_width = $max;
-			$dest_height = intval(($height * $max) / $width);
-		} else {
-			$dest_width = intval(($width * $max) / $height);
-			$dest_height = $max;
+	/**
+	 * @param integer $width  width
+	 * @param integer $height height
+	 * @param integer $max    max
+	 * @return array
+	 */
+	public function scaleImageTo($width, $height, $max)
+	{
+		$dest_width = $dest_height = 0;
+	
+		if ((!$width) || (!$height)) {
+			return false;
 		}
-	} else {
-		if ($width > $max) {
-			$dest_width = $max;
-			$dest_height = intval(($height * $max) / $width);
+	
+		if ($width > $max && $height > $max) {
+			// very tall image (greater than 16:9)
+			// constrain the width - let the height float.
+	
+			if ((($height * 9) / 16) > $width) {
+				$dest_width = $max;
+				$dest_height = intval(($height * $max) / $width);
+			} elseif ($width > $height) {
+				// else constrain both dimensions
+				$dest_width = $max;
+				$dest_height = intval(($height * $max) / $width);
+			} else {
+				$dest_width = intval(($width * $max) / $height);
+				$dest_height = $max;
+			}
 		} else {
-			if ($height > $max) {
-
-				// very tall image (greater than 16:9)
-				// but width is OK - don't do anything
-
-				if ((($height * 9) / 16) > $width) {
+			if ($width > $max) {
+				$dest_width = $max;
+				$dest_height = intval(($height * $max) / $width);
+			} else {
+				if ($height > $max) {
+					// very tall image (greater than 16:9)
+					// but width is OK - don't do anything
+	
+					if ((($height * 9) / 16) > $width) {
+						$dest_width = $width;
+						$dest_height = $height;
+					} else {
+						$dest_width = intval(($width * $max) / $height);
+						$dest_height = $max;
+					}
+				} else {
 					$dest_width = $width;
 					$dest_height = $height;
-				} else {
-					$dest_width = intval(($width * $max) / $height);
-					$dest_height = $max;
 				}
-			} else {
-				$dest_width = $width;
-				$dest_height = $height;
-			}
-		}
-	}
-	return array("width" => $dest_width, "height" => $dest_height);
-}
-
-function store_photo(App $a, $uid, $imagedata = "", $url = "") {
```

Comment:
You removed this function, but you haven't changed the corresponding calls to this function in the file ```fromgplus.php```. Please make sure that either the call is changed or this function is restored. And no: I don't know the difference between the two functions that store photos.